src/ring_theory/local_properties.lean

```diff
@@ -270,6 +277,116 @@ end
 
 end ideal
 
+section module
+
+variables {X Y : Type*}
+variables [add_comm_group X] [module R X] [add_comm_group Y] [module R Y] (g : X →ₗ[R] Y)
+
+lemma submodule.le_of_localization_maximal (M M' : submodule R X)
+  (H : ∀ (J : ideal R) (hJ : J.is_maximal),
+    by exactI M.localize J.prime_compl ≤ M'.localize J.prime_compl) : M ≤ M' :=
+begin
+  intros x hx,
+  by_contradiction hx',
+  have : (submodule.span R {M'.mkq x}).annihilator ≠ ⊤,
+  { rwa [ne.def, submodule.annihilator_eq_top_iff, submodule.span_singleton_eq_bot,
+      ← linear_map.mem_ker, submodule.ker_mkq] },
+  obtain ⟨p, hp₁, hp₂⟩ := ideal.exists_le_maximal _ this,
+  specialize H p hp₁,
+  obtain ⟨y, hy, ⟨s, hs⟩, e⟩ := H (submodule.mem_localize_of_mem hx 1),
+  obtain ⟨⟨u, hu⟩, e⟩ := localized_module.mk_eq.mp e,
+  simp only [submonoid.smul_def, one_smul, smul_zero, subtype.coe_mk, @eq_comm X 0] at e,
+  exact (mul_mem hu hs) (hp₂ $ by { rw [submodule.mem_annihilator_span_singleton, ← M'.mkq.map_smul,
+    ← linear_map.mem_ker, submodule.ker_mkq, mul_smul, e], exact M'.smul_mem _ hy })
```

Comment:
If we're in tactic mode anyway, it seems much more readable to break this into multiple tactic calls.
```suggestion
  refine (mul_mem hu hs) (hp₂ _),
  rw [submodule.mem_annihilator_span_singleton, ← M'.mkq.map_smul,
    ← linear_map.mem_ker, submodule.ker_mkq, mul_smul, e],
  exact M'.smul_mem _ hy
```